compileopts/target.go

```diff
@@ -179,7 +179,15 @@ func (spec *TargetSpec) resolveInherits() error {
 // Load a target specification.
 func LoadTarget(options *Options) (*TargetSpec, error) {
 	if options.Target == "" {
-		return defaultTarget(options)
+		if options.GOARCH == "wasm" && options.GOOS == "js" {
+			options.Target = "wasm"
+		} else if options.GOARCH == "wasm" && options.GOOS == "wasip1" {
+			options.Target = "wasip1"
+		} else if options.GOARCH == "wasm" && options.GOOS == "wasip2" {
```

Comment:
Will setting `GOOS=wasip2` even work? Will Go choke on the unknown `GOOS`?

Reply:
Yes, it will work. I've tested it. GOOS is rewritten to what is set in targets/wasip2.json before `go` commands are invoked.